applications/GeoMechanicsApplication/custom_constitutive/linear_elastic_plane_strain_2D_law.cpp

```diff
@@ -89,12 +101,83 @@ void GeoLinearElasticPlaneStrain2DLaw::CalculatePK2Stress(const Vector& rStrainV
-    // Total formulation
-    noalias(rStressVector) = prod(C, rStrainVector);
+
+    // Incremental formulation
+    noalias(mStressVector) = mStressVectorFinalized + prod(C, mDeltaStrainVector);
+
+    SetExternalStressVector(rStressVector);
 
     KRATOS_CATCH("")
 }
 
+void GeoLinearElasticPlaneStrain2DLaw::UpdateInternalDeltaStrainVector(const Vector& r_strain_vector){
+    mDeltaStrainVector = r_strain_vector - mStrainVectorFinalized;
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::UpdateInternalStrainVectorFinalized(const Vector& r_strain_vector) {
+    this->SetInternalStrainVector(r_strain_vector);
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::SetExternalStressVector(Vector& r_stress_vector) const{
+    r_stress_vector = mStressVector;
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::SetInternalStressVector(const Vector& r_stress_vector){
+    mStressVectorFinalized = r_stress_vector;
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::SetInternalStrainVector(const Vector& r_strain_vector){
+    mStrainVectorFinalized = r_strain_vector;
+}
+void GeoLinearElasticPlaneStrain2DLaw::InitializeMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
+{
+    KRATOS_TRY
+    if (!mIsModelInitialized)
+    {
+        // stress and strain vectors must be initialized:
+        const Vector& r_stress_vector = rValues.GetStressVector();
+        const Vector& r_strain_vector = rValues.GetStrainVector();
+
+        SetInternalStressVector(r_stress_vector);
+        SetInternalStrainVector(r_strain_vector);
+
+        mIsModelInitialized = true;
+    }
+    KRATOS_CATCH("")
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::FinalizeMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
+{
+    UpdateInternalStrainVectorFinalized(rValues.GetStrainVector());
+    mStressVectorFinalized = mStressVector;
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::FinalizeMaterialResponsePK2(ConstitutiveLaw::Parameters& rValues)
+{
+    // Small deformation so we can call the Cauchy method
+    FinalizeMaterialResponseCauchy(rValues);
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::save(Serializer &rSerializer) const {
+    KRATOS_SERIALIZE_SAVE_BASE_CLASS( rSerializer, LinearPlaneStrainK0Law)
+    rSerializer.save("StressVector",mStressVector);
+    rSerializer.save("StressVectorFinalized",mStressVectorFinalized);
+    rSerializer.save("DeltaStrainVector",mDeltaStrainVector);
+    rSerializer.save("StrainVectorFinalized",mStrainVectorFinalized);
+    rSerializer.save("mIsModelInitialized",mIsModelInitialized);
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::load(Serializer &rSerializer) {
+    KRATOS_SERIALIZE_LOAD_BASE_CLASS( rSerializer, LinearPlaneStrainK0Law)
+    rSerializer.load("StressVector",mStressVector);
+    rSerializer.load("StressVectorFinalized",mStressVectorFinalized);
+    rSerializer.load("DeltaStrainVector",mDeltaStrainVector);
+    rSerializer.load("StrainVectorFinalized",mStrainVectorFinalized);
+    rSerializer.load("mIsModelInitialized",mIsModelInitialized);
```

Comment:
And the same nitpicking here:
```suggestion
    rSerializer.load("StressVector", mStressVector);
    rSerializer.load("StressVectorFinalized", mStressVectorFinalized);
    rSerializer.load("DeltaStrainVector", mDeltaStrainVector);
    rSerializer.load("StrainVectorFinalized", mStrainVectorFinalized);
    rSerializer.load("mIsModelInitialized", mIsModelInitialized);
```

Reply:
done